### PR TITLE
Docker Development Improvements

### PR DESCRIPTION
**Pull request by beninato8**
This PR is a bit all over the place. My original goal was to understand how rankings worked, but I kept running into other things.

I was looking at the Season 13 Updates, and saw for Silph Factions (Master League), Togekiss is ranked 171 using Air Slash as a fast move. This didn't make sense to me, as when I did a Matrix battle to compare Charm and Air Slash against the meta, Air Slash as a record of 8-35-0, and Charm has a record of 17-25-1.

I went to understand how the rankings worked by checking out `localhost/ranker.php`. However, I was getting an error where I did not have permission to write the output files. I've been using docker since I didn't want to deal with XAMPP again. My solution to write files was to sync files between the host and the container by adding a volume to `docker-compose.yml`. This also has the benefit of code changes updating upon refreshing the pages, instead of having to start and stop Docker and copy over files to the container, or copy files over to `htdocs` directory as I did with XAMPP previously. And rankings are just generated in the rankings directory in the GitHub repo, so no need to worry about having 2 areas. Finally, since the URL now contains `src`, `header.php` will generate a random version number to prevent caching, and allow access to the dev panel. 

Speaking of the dev panel...it's pretty neat! I hadn't found it until today though. I was looking at the browser dev tools, I saw a few errors. The first was an issue with not being able to find `http://localhost/pvpoke/src/articles/article-assets/developer-notes-1-27-0/thumb.jpg`. I found this was from `css`, and discovered that there was a whole dev tools panel for PvPoke. However, since `article-assets` was in the `.gitignore`, the file path to the wrench did not exist. So I added that image so I could use the dev tools pannel.

The next error in dev tools was I kept getting `XML Parsing Error: syntax error` at `Line Number 1, Column 1` for reading in JSON files. Apparently reading local files causes that error to show up, so the solution from [here](https://stackoverflow.com/a/18292092/7203518) that worked was to specify `mimeType: "application/json"`. To set the `mimeType`, I replaced `$.getJSON` with an equivalent `$.ajax` call from [here](https://api.jquery.com/jQuery.getJSON/#jQuery-getJSON-url-data-success). This could be done in many more places, but I only added it to a few of the ones I ran into.

I also added a few other smaller changes that I ran into, such as:
 - Areas where Pokémon was missing the accent
 - Adding specific error messages for `write.php`
 - Removing duplicate lines in `.gitignore`

Not related to this specific pull request, but I've been wondering if there are other resources than the README for people who would like to contribute. For example, I was curious if there was a specific style guide for the project. I saw there was a dead link to a Discord server here: https://github.com/pvpoke/pvpoke/pull/164#issuecomment-914635724

Now that I have a little bit of a better understanding of how things work, I'd like to get back to my original goal of looking into rankings before I got a bit sidetracked. While I was looking at the code, I was wondering where the overrides come from. Are they manually determined? If so, how? Same with the weights. One thing that has always bothered me was ML Garchomp being recommended with Mud Shot, Outrage, and Earth Power when I personally prefer Dragon Tail, Outrage, and Sand Tomb (and it seems to perform better if I compare 1 shields performance vs the ML Meta).

### CODE REVIEW

File: src/js/GameMaster.js [hunk 1]
```diff
@@ -28,47 +28,56 @@ var GameMaster = (function () {
 			gmVersion = "gamemaster.min";
 		}
 
-		$.getJSON( webRoot+"data/"+gmVersion+".json?v="+siteVersion, function( data ){
```
Comment (beninato8) [inline src/js/GameMaster.js:31]: Aside from the change from `$.getJSON` to `$.ajax`, this is all just whitespace changes - see https://github.com/pvpoke/pvpoke/pull/198/files?diff=unified&w=1

File: src/header.php [hunk 2]
```diff
@@ -80,13 +80,13 @@
 <meta name="viewport" content="width=device-width, initial-scale=1.0">
 <?php
 if(! isset($META_TITLE)){
-	$META_TITLE = 'PvPoke | Open-Source Battle Simulator, Rankings &amp; Team Building for Pokemon GO PvP';
+	$META_TITLE = 'PvPoke | Open-Source Battle Simulator, Rankings &amp; Team Building for Pokémon GO PvP';
```
Comment (pvpoke) [inline src/header.php:83]: For style/brand reasons, I'd like to keep the word "Pokemon" without the accent throughout the site.

**Review by pvpoke — CHANGES_REQUESTED** [review]
Thanks for these changes! Everything looks good except I'd like to keep the word "Pokemon" without the accent throughout the site. Let me know when you have a chance to revert this and I'll get these merged!

**Comment by beninato8** [comment]
Hey @pvpoke I undid the changes with the accent. Does it still look OK to you?

**Review by pvpoke — APPROVED** [review]

**Pull request merged**